File: backend/finan-exp-backend/Categories/Models/Category.cs
using Shared.Classes;
using Shared.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Categories.Models
{
  [Table("Categories")]
  public class Category : FullEntity, ICategoryModel
  {
    [Required]
    public string Name { get; set; }
    [Required]
    public TransactionType TransactionType { get; set; }

    [Required]
    public bool IsArchived { get; set; }

  }
}

File: src/app/core/models/categories/category.ts
import { TransactionType } from './../../../shared/support/enums/transactionTypes/transaction-types';
import { FullEntity } from 'finan-exp-sdk';

export abstract class Category extends FullEntity {
  public name!: string;

  public transactionType!: TransactionType;

  public isArchived!: boolean;
}
